File: apps/android/core/model/src/main/kotlin/com/sigmundgranaas/turbo/expressive/domain/MapCollection.kt
package com.sigmundgranaas.turbo.expressive.domain

/**
 * A user-defined grouping of map entities (markers, tracks). Local-only — a
 * lightweight folder with a colour + optional icon and a membership count.
 */
data class MapCollection(
    val id: String,
    val name: String,
    val colorArgb: Long? = null,
    val icon: String? = null,
    val itemCount: Int = 0,
)

/** The kinds of entity a collection can contain. */
enum class CollectionItemType { Marker, Path }

File: apps/ios/Sources/CoreModel/MapCollection.swift
import Foundation

/// A user-defined grouping of map entities (markers, tracks). A lightweight
/// folder with a colour + optional icon and a membership count. Mirrors
/// `domain.MapCollection`.
public struct MapCollection: Identifiable, Hashable, Sendable {
    public let id: String
    public let name: String
    public let colorArgb: Int64?
    public let icon: String?
    public let itemCount: Int

    public init(id: String, name: String, colorArgb: Int64? = nil, icon: String? = nil, itemCount: Int = 0) {
        self.id = id
        self.name = name
        self.colorArgb = colorArgb
        self.icon = icon
        self.itemCount = itemCount
    }
}

/// The kinds of entity a collection can contain. Mirrors `domain.CollectionItemType`.
public enum CollectionItemType: String, Sendable, Codable {
    case marker, path
}
